packages/tools/kadena-cli/src/account/utils/__tests__/addHelpers.test.ts

```diff
@@ -0,0 +1,358 @@
+import * as inquirerPrompts from '@inquirer/prompts';
+import yaml from 'js-yaml';
+import { http, HttpResponse } from 'msw';
+import path from 'path';
+
+import type { ChainId } from '@kadena/types';
+import { defaultAccountPath } from '../../../constants/account.js';
+import { server } from '../../../mocks/server.js';
+import type { INetworkCreateOptions } from '../../../networks/utils/networkHelpers.js';
+import { services } from '../../../services/index.js';
+import type { IAddAccountManualConfig, Predicate } from '../../types.js';
+import {
+  compareAndUpdateConfig,
+  createAccount,
+  createAccountName,
+  getAccountDetailsFromChain,
+  validateAccountDetails,
+  validatePublicKeys,
+  writeConfigInFile,
+} from '../addHelpers.js';
+
+const networkConfig: INetworkCreateOptions = {
+  networkHost: 'https://api.testnet.chainweb.com',
+  networkExplorerUrl: 'https://explorer.chainweb.com/testnet04',
+  networkId: 'testnet04',
+  network: 'testnet',
+};
+
+const defaultConfig: IAddAccountManualConfig = {
+  accountAlias: 'accountAlias',
+  fungible: 'coin',
+  publicKeysConfig: [],
+  predicate: 'keys-all' as Predicate,
+  ...networkConfig,
+  networkConfig,
+  chainId: '1' as ChainId,
+};
+
+describe('createAccountName', () => {
+  afterEach(() => {
+    server.resetHandlers();
+  });
+
+  it('should return undefined if no public keys are provided', async () => {
+    const accountName = await createAccountName(defaultConfig);
+    expect(accountName).toBeUndefined();
+  });
+
+  it('should return undefined when predicate is not provided', async () => {
+    const accountName = await createAccountName({
+      ...defaultConfig,
+      predicate: undefined as unknown as Predicate,
+      publicKeysConfig: ['publicKey1', 'publicKey2'],
+    });
+    expect(accountName).toBeUndefined();
+  });
```

Comment:
this test should not be needed. trust typescript, and if input comes from the user use zod as early as possible to validate it